sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DistributionAndOrderingUtils.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.v2
+
+import org.apache.spark.sql.{catalyst, AnalysisException}
+import org.apache.spark.sql.catalyst.analysis.Resolver
+import org.apache.spark.sql.catalyst.expressions.{NamedExpression, SortOrder}
+import org.apache.spark.sql.catalyst.plans.logical.{LogicalPlan, RepartitionByExpression, Sort}
+import org.apache.spark.sql.connector.distributions.{ClusteredDistribution, OrderedDistribution, UnspecifiedDistribution}
+import org.apache.spark.sql.connector.expressions.{Expression, FieldReference, IdentityTransform, NullOrdering, SortDirection, SortValue}
```

Comment:
nit: `import ...{Expression => V2 Expression, ...}` to avoid name conflicts.

Reply:
Oh, cool. I see we use this idea in `AstBuilder`.

@cloud-fan, do you think we should keep using `catalyst.expressions.Expression` or just `Expression` once we add an import for `V2Expression`?

Reply:
Let's use `Expression` and `V2Expression` in this file, to be consistent with other places. `Expression` is widely used in spark SQL code base and by default people will think it's the catalyst `Expression`.

Reply:
Sounds good, will update the PR.

Reply:
@cloud-fan, when doing this change, I realized we also have things like `SortDirection` both in the connector package as well as an internal one. What shall we do about that? Should it become `V2SortDirection` or something else?

Reply:
sorry I missed this message, yea we should rename it to `V2SortDirection` in the import.

Reply:
if there are too many things to rename, we can do `import org.apache.spark.sql.connector.{expressions => v2}`, and use `v2.Expression` to avoid conflicts with catalyst `Expression`

Reply:
I'll try and see what makes more sense. Thanks, @cloud-fan.

Reply:
I went with `V2SortDirection`. Resolving this thread.